python/pyspark/ml/tests.py

```diff
@@ -230,6 +230,15 @@ def test_pipeline(self):
         self.assertEqual(5, transformer3.dataset_index)
         self.assertEqual(6, dataset.index)
 
+    def test_identity_pipeline(self):
+        dataset = MockDataset()
+
+        def doTransform(pipeline):
+            pipeline_model = pipeline.fit(dataset)
+            return pipeline_model.transform(dataset)
+        self.assertEqual(dataset.index, doTransform(Pipeline()).index)
+        self.assertEqual(dataset.index, doTransform(Pipeline(stages=[])).index)
```

Comment:
Should we also check that `setParams(stages=[])` and `Pipeline().getStages()` return the expected value?